src/FakeItEasy/Configuration/RuleBuilder.cs

```diff
@@ -148,9 +159,7 @@ public IAfterCallSpecifiedConfiguration CallsBaseMethod()
 
             public IReturnValueConfiguration<TMember> WhenArgumentsMatch(Func<ArgumentCollection, bool> argumentsPredicate)
             {
-                Guard.AgainstNull(argumentsPredicate, nameof(argumentsPredicate));
-
-                this.ParentConfiguration.RuleBeingBuilt.UsePredicateToValidateArguments(argumentsPredicate);
+                this.ParentConfiguration.WhenArgumentsMatch(argumentsPredicate);

```

Comment:
I eventually put it in another commit and cherry picked it to another branch; I'll submit a separate PR later.